data-products/src/main/scala/org/sunbird/analytics/job/report/AssessmentMetricsJob.scala

```diff
@@ -75,58 +75,50 @@ object AssessmentMetricsJob extends optional.Application with IJob with BaseRepo
-    val userCoursesDF = loadData(spark, Map("table" -> "user_courses", "keyspace" -> sunbirdCoursesKeyspace))
+    val courseBatchDF = loadData(spark, Map("table" -> "course_batch", "keyspace" -> sunbirdCoursesKeyspace),"org.apache.spark.sql.cassandra").select("courseid", "batchid", "startdate", "enddate")
+    val userCoursesDF = loadData(spark, Map("table" -> "user_courses", "keyspace" -> sunbirdCoursesKeyspace),"org.apache.spark.sql.cassandra")
       .filter(lower(col("active")).equalTo("true"))
       .select(col("batchid"), col("userid"), col("courseid"), col("active")
         , col("completionpercentage"), col("enrolleddate"), col("completedon"))
-    val userDF = loadData(spark, Map("table" -> "user", "keyspace" -> sunbirdKeyspace)).select(col("userid"),
-      col("maskedemail"),
-      col("firstname"),
-      col("lastname"),
-      col("maskedphone"),
-      col("rootorgid"),
-      col("locationids"),
-      col("channel")
-    ).persist()
-    val userOrgDF = loadData(spark, Map("table" -> "user_org", "keyspace" -> sunbirdKeyspace)).filter(lower(col("isdeleted")) === "false").select(col("userid"), col("organisationid")).persist()
-    val organisationDF = loadData(spark, Map("table" -> "organisation", "keyspace" -> sunbirdKeyspace))
-      .select(col("id"), col("orgname"), col("channel"), col("isrootorg"), col("locationids"), col("orgcode")).persist()
-    val locationDF = loadData(spark, Map("table" -> "location", "keyspace" -> sunbirdKeyspace))
-      .select(col("id"), col("name"), col("type")).persist()
-    val externalIdentityDF = loadData(spark, Map("table" -> "usr_external_identity", "keyspace" -> sunbirdKeyspace))
-      .select(col("provider"), col("idtype"), col("externalid"), col("userid")).persist()
-    val assessmentProfileDF = loadData(spark, Map("table" -> "assessment_aggregator", "keyspace" -> sunbirdCoursesKeyspace))
+
+    val userDF = loadData(spark, Map("keys.pattern" -> "*","infer.schema" -> "true"),"org.apache.spark.sql.redis")
+      .select(col("userid"),col("firstname"),col("lastname"),col("maskedemail"),col("maskedphone"),
+        col("districtname"), col("externalid"),col("schoolname"),col("schooludisecode"),col("statename"),col("orgname"),
+        concat_ws(" ", col("firstname"), col("lastname")).as("username"))
+
+    val assessmentProfileDF = loadData(spark, Map("table" -> "assessment_aggregator", "keyspace" -> sunbirdCoursesKeyspace),"org.apache.spark.sql.cassandra")
       .select("course_id", "batch_id", "user_id", "content_id", "total_max_score", "total_score", "grand_total")
 
     implicit val sqlContext = new SQLContext(spark.sparkContext)
     import sqlContext.implicits._
 
-    val druidResult = DruidDataFetcher.getDruidData(druidQuery)
-    val finalResult = druidResult.map { f => JSONUtils.deserialize[DruidOutput](f) }
-    val finalDF = finalResult.toDF()
+    val courseChannelDenormDF = courseBatchDF.collect().map(row => {
```

Comment:
Collecting the entire data frame onto the driver is not a good practice. I would like to know if we can avoid doing a collect.